bagit.py

```diff
@@ -162,30 +163,41 @@ def make_bag(
     if checksums is None:
         checksums = DEFAULT_CHECKSUMS
 
-    bag_dir = os.path.abspath(bag_dir)
+    if dest_dir:
+        bag_name = os.path.basename(bag_dir)
+        dest_dir = os.path.abspath(os.path.join(dest_dir, bag_name))
+        if not os.path.isdir(dest_dir):
+            os.makedirs(dest_dir)
+        else:
+            raise RuntimeError(_("The following directory already exists:\n%s"), dest_dir)
```

Comment:
Consider allowing an empty directory as a target, useful e.g. when creating that directory requires different permissions than the user running bagit has.

Reply:
Good point. Would checking for an empty directory like this be enough?
```suggestion
      elif len(os.listdir(dest_dir)) > 0:
            raise RuntimeError(_("The following directory already exists and contains files:\n%s"), dest_dir)
```

Reply:
That will work because `os.listdir` is documented as not including `.` or `..` but I was wondering about performance if someone points it at a directory which a large number of files — the classic Unix answer being `os.stat(dest_dir).st_nlink > 2` — but I don't think that'll really be an issue in this scenario.